src/Compilers/CSharp/Portable/Binder/FixedStatementBinder.cs

```diff
@@ -42,14 +42,19 @@ override protected ImmutableArray<LocalSymbol> BuildLocals()
             return ImmutableArray<LocalSymbol>.Empty;
         }
 
-        internal override ImmutableArray<LocalSymbol> GetDeclaredLocalsForScope(CSharpSyntaxNode node)
+        internal override ImmutableArray<LocalSymbol> GetDeclaredLocalsForScope(CSharpSyntaxNode scopeDesignator)
         {
-            return this.Locals;
+            if (_syntax == scopeDesignator)
```

Comment:
This condition looks a lot like a release assert. 
Perhaps mismatched syntax should be debug asserts?